src/drivers/vga/text_mode.rs

```diff
@@ -101,10 +83,15 @@ impl Writer {
 
         self.handle_scrolling();
 
+        let color_char = self.state.ransid_process(ch);
+        if color_char.ascii == b'\0' {
```

Comment:
Can you make `ransid_process` return an `Option<ColorChar>` instead of a null byte? intent is clearer that way

Reply:
Also the processing should be done before the match and `handle_scrolling` because otherwise it might scroll, while `ransid_process` returns a null byte and nothing is printed